### PR TITLE
refactor(servers): improve postgres error message

### PR DESCRIPTION
**Pull request by leaf-potato**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/4340

## What's changed and what's your intention?

- Define enum `PgErrorCode` struct for pgwire.
- Impl `from` trait for mapping `StatusCode` to `PgErrorCode`.
- Impl `to_err_info` to trans PgErrorCode to ErrorInfo.
- Use enum `PgErrorCode` intead of `ErrorInfo::new()`. 

before:
```rust
match output {
    Ok(o) => todo!(),
    Err(e) => Ok(Response::Error(Box::new(ErrorInfo::new(
        "ERROR".to_string(),
        "XX000".to_string(),
        e.output_msg(),
    ))))
}
```
after:
```rust
match output {
    Ok(o) => todo!(),
    Err(e) => Ok(Response::Error(Box::new(
        PgErrorCode::from(e.status_code()).to_err_info(e.output_msg())
    )))
}
```

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
A great work! Thanks a lot.

**Comment by killme2008** [comment]
@sunng87 @waynexia Please take a look.

File: src/servers/src/postgres/types/error.rs [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+    // === Begin of Class XX — Internal Error ===
+    /// internal_error
+    #[snafu(display("internal_error"))]
+    EcXX000 = 4200,
+    // === End of Class XX — Internal Error =====
+}
+
+impl PgErrorCode {
+    fn severity(&self) -> ErrorSeverity {
+        match self {
```
Comment (sunng87) [inline src/servers/src/postgres/types/error.rs:340]: Is there any reference for this mapping?

File: src/servers/src/postgres/types/error.rs [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+// limitations under the License.
+
+use common_error::status_code::StatusCode;
+use pgwire::error::ErrorInfo;
+use snafu::Snafu;
+use strum::{AsRefStr, Display, EnumIter, EnumMessage};
+
+#[derive(Display, Debug, PartialEq)]
+#[allow(dead_code)]
+enum ErrorSeverity {
```
Comment (sunng87) [inline src/servers/src/postgres/types/error.rs:22]: why not make this pub

File: src/servers/src/postgres/types/error.rs [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+    #[strum(serialize = "INFO")]
+    Info = 0,
```
Comment (sunng87) [inline src/servers/src/postgres/types/error.rs:24]: I think we don't need to provide numerical representation for this enum. The postgres protocol doesn't define it

File: src/servers/src/postgres/types/error.rs (hunk 1 above)
Comment (sunng87) [reply src/servers/src/postgres/types/error.rs:340]: I see we are mapping them based on semantics. It sounds good at least for now.

**Review by sunng87 — APPROVED** [review]

File: src/servers/src/postgres/types/error.rs [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+    Warning,
+    #[strum(serialize = "ERROR")]
+    Error,
+    #[strum(serialize = "FATAL")]
+    Fatal,
+    #[strum(serialize = "PANIC")]
+    Panic,
+}
+
+// refer to: https://www.postgresql.org/docs/16/errcodes-appendix.html
```
Comment (waynexia) [inline src/servers/src/postgres/types/error.rs:39]: 💯 nice link

**Pull request merged**